src/test/checkqueue_tests.cpp

```diff
@@ -373,71 +347,7 @@ BOOST_AUTO_TEST_CASE(test_CheckQueue_FrozenCleanup)
     FrozenCleanupCheck::cv.notify_one();
     // Wait for control to finish
     t0.join();
-    tg.interrupt_all();
-    tg.join_all();
     BOOST_REQUIRE(!fails);
-}
-
-
-/** Test that CCheckQueueControl is threadsafe */
-BOOST_AUTO_TEST_CASE(test_CheckQueueControl_Locks)
-{
-    auto queue = MakeUnique<Standard_Queue>(QUEUE_BATCH_SIZE);
```

Comment:
d64dd5ffee064b0b943b7e00b16b0ad28ac65474: Why remove this test?

Reply:
I have no idea how to pass a customized callback into the `CCheckQueue::StartWorkerThreads(int)`.

Reply:
Usually you want to use a callable template parameter...
```c++
template<typename Callable>
function(int x, Callable&& function) {
    function(x);
}
function(1, [](int x) { return x});
function(1, [](int x) { });
```

Reply:
@JeremyRubin Thank you for your suggestion :)

Sorry for my broken English, but I meant do we really need a unit test if the `CCheckQueue` class has no interface to test?

Reply:
Sorry to just be responding on this:

I think if there isn't an interface for something, but you can expose one to make testing more robust, it's worthwhile.

E.g., the CheckQueue does not need to be templated as we know a concrete single type for it, but templating it allows us to mock out the job type and probe the internal execution, which is a good thing.

What's the missing interface here? I don't see one being used in the code for this test.

Reply:
@MarcoFalke @JeremyRubin 
I was confused about this test (I don't know the exact reason of that confusion now).
Test has been [restored](https://github.com/bitcoin/bitcoin/pull/18710#issuecomment-639992778).